File: h.py
import cv2
import numpy as np
import argparse

ap = argparse.ArgumentParser()
ap.add_argument("-i","--image", help = 'required image path file')
args = vars(ap.parse_args())


def onTrackbarChange(max_slider):
	global img
	global dst
	global gray
	dst = np.copy(gray)
	edges = cv2.Canny(gray, 75, 150)
	lines = cv2.HoughLinesP(edges, 1, np.pi/180, max_slider, maxLineGap=250)
	
	for line in lines:
		x1, y1, x2, y2 = line[0]
		cv2.line(dst, (x1, y1), (x2, y2), (0, 255, 0), 3)

	# cv2.imshow("edges", edges)
	cv2.imshow("frame",dst)
	


if __name__ == "__main__":
	img = cv2.imread(args["image"])
	dst = np.copy(img)
	cv2.namedWindow("frame")

	gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
	
	thresh = 80
	val = 100

	cv2.createTrackbar("threshold", "frame", thresh, val, onTrackbarChange)
	onTrackbarChange(val)

	while True:
		cv2.imshow("source image",img)
		key = cv2.waitKey(1)
		if key == ord('q'):
			break

	cv2.destroyAllWindows()

File: h.cpp
#include <opencv2/core/core.hpp>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/video/background_segm.hpp>
#include <opencv2/opencv.hpp>
#include <stdio.h>
#include <iostream>

using namespace cv;
using namespace std;
Mat dst, cdst, cdstP;
int thresh;
const int thresh_max = 100;
double t;
 vector<Vec4i> linesP;

void on_trackbar( int , void* )

{
  
  cdstP = cdst.clone();
  HoughLinesP(dst, linesP, 1, CV_PI/180, thresh, 50, 10 );
 
   for( size_t i = 0; i < linesP.size(); i++ )
    {
        Vec4i l = linesP[i];
        line( cdstP, Point(l[0], l[1]), Point(l[2], l[3]), Scalar(0,0,255), 3, LINE_AA);
    }
   imshow( "Linear Blend", cdstP );

}


int main(int argc, char** argv){

  const char* file = argv[1];
  Mat src = imread(file, 0);
  if(src.empty())
  {
    cout << "Error reading image" << file<< endl;
    return -1;
  }


  Canny(src, dst, 50, 200, 3);
  cvtColor(dst, cdst, COLOR_GRAY2BGR); // Convert to BGR
  cdstP = cdst.clone();

   // will hold the results of the detection
  namedWindow("Linear Blend", 1);

 
 
  thresh = 50;
  char TrackbarName[50];
  sprintf( TrackbarName, "Alpha x %d", thresh_max);

  createTrackbar("threshold", "Linear Blend", &thresh,thresh_max,on_trackbar);

  on_trackbar(thresh,0);

  imshow("source", src);
  // imshow("detected lines", cdstP);
  while(true)
  {
    int c;
    c = waitKey( 20 );
    if( (char)c == 27 )
      { break; }
  }
  //waitKey();
  //return 0;

}
